torchvision/models/quantization/utils.py

```diff
@@ -18,28 +19,36 @@ def _replace_relu(module: nn.Module) -> None:
         module._modules[key] = value
 
 
-def quantize_model(model: nn.Module, backend: str) -> None:
+def quantize_model(model: nn.Module, backend: str, fx_mode: bool) -> None:
     _dummy_input_data = torch.rand(1, 3, 299, 299)
     if backend not in torch.backends.quantized.supported_engines:
         raise RuntimeError("Quantized backend not supported ")
     torch.backends.quantized.engine = backend
     model.eval()
     # Make sure that weight qconfig matches that of the serialized models
     if backend == "fbgemm":
-        model.qconfig = torch.ao.quantization.QConfig(  # type: ignore[assignment]
+        qconfig = torch.ao.quantization.QConfig(  # type: ignore[assignment]
             activation=torch.ao.quantization.default_observer,
             weight=torch.ao.quantization.default_per_channel_weight_observer,
         )
     elif backend == "qnnpack":
-        model.qconfig = torch.ao.quantization.QConfig(  # type: ignore[assignment]
+        qconfig = torch.ao.quantization.QConfig(  # type: ignore[assignment]
             activation=torch.ao.quantization.default_observer, weight=torch.ao.quantization.default_weight_observer
         )
 
-    # TODO https://github.com/pytorch/vision/pull/4232#pullrequestreview-730461659
-    model.fuse_model()  # type: ignore[operator]
-    torch.ao.quantization.prepare(model, inplace=True)
-    model(_dummy_input_data)
-    torch.ao.quantization.convert(model, inplace=True)
+    if fx_mode:
+        qconfig_dict = {"": qconfig}
+        model = torch.ao.quantization.quantize_fx.fuse_fx(model)
```

Comment:
I thought the fusion happens inside of prepare, do we need to do it again here?

Reply:
yeah we don't need to call fuse_fx explicitly I think

Reply:
Ok, will remove